libcontainer/cgroups/utils.go

```diff
@@ -55,12 +55,11 @@ func IsCgroup2HybridMode() bool {
 		var st unix.Statfs_t
 		err := unix.Statfs(hybridMountpoint, &st)
 		if err != nil {
-			if os.IsNotExist(err) {
-				// ignore the "not found" error
-				isHybrid = false
-				return
+			isHybrid = false
+			if !os.IsNotExist(err) {
+				// Report unexpected errors.
+				logrus.WithError(err).Debugf("statfs(%q) failed", hybridMountpoint)
 			}
-			panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
```

Comment:
still need to `return` rather than continue to `isHybrid = st.Type == unix.CGROUP2_SUPER_MAGIC` in the error case, right?

Reply:
Right. Updated; thanks for catching this!

I am also on the verge of removing `isHybrid = false` line since it does nothing (since isHybrid is false by default), but I guess it is better to leave it as is it for the sake of readability.